bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -111,15 +139,33 @@ private SecurityContext authenticateBearerToken(String token) throws Authenticat
         }
     }
 
-    private SecurityContext authenticateUsernamePassword(String username, String password)
+    private SecurityContext authenticateUsernamePassword(String username, String password, String authCacheKey)
             throws AuthenticationException {
+        final ByteBuffer cacheKey = getCacheKey(authCacheKey);
+        final UserSecurityContext cachedValue = AuthFilter.authCache.get(cacheKey);
+        if (cachedValue != null) {
+            return cachedValue;
+        }
+
         UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(username, password);
         Authentication auth = userRegistry.authenticate(credentials);
```

Comment:
So if there was some entry in the cache, we fully skip the authentication? Same question as above: How do you ensure that the cache entry is still valid? How are entries invalidated if changes to the userRegistry are done?